nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/test/java/org/apache/hive/streaming/TestNiFiRecordSerDe.java

```diff
@@ -0,0 +1,360 @@
+                        new RecordField("longc", RecordFieldType.LONG.getDataType()),
+                        new RecordField("boolc", RecordFieldType.BOOLEAN.getDataType()),
+                        new RecordField("floatc", RecordFieldType.FLOAT.getDataType()),
+                        new RecordField("doublec", RecordFieldType.DOUBLE.getDataType()),
+                        new RecordField("stringc", RecordFieldType.STRING.getDataType()),
+                        new RecordField("varcharc", RecordFieldType.STRING.getDataType()),
+                        new RecordField("charc", RecordFieldType.CHAR.getDataType()),
+                        new RecordField("binaryc", RecordFieldType.ARRAY.getArrayDataType(RecordFieldType.BYTE.getDataType())),
+                        new RecordField("datec", RecordFieldType.DATE.getDataType("yyyy-MM-dd")),
+                        new RecordField("timestampc", RecordFieldType.TIMESTAMP.getDataType("yyyy-MM-dd HH:mm:ss")),
+                        new RecordField("decimalc", RecordFieldType.DOUBLE.getDataType())
+                )
+        );
+
+        long currentTimeMillis = System.currentTimeMillis();
+
+        Object deserialized = serDe.deserialize(new ObjectWritable(new MapRecord(schema, new HashMap(){
+            {
+                put("bytec", Byte.valueOf((byte)2));
+                put("shortc", Short.valueOf((short)45));
+                put("intc", Integer.valueOf(95));
+                put("longc", Long.valueOf(876L));
+                put("boolc", Boolean.TRUE);
+                put("floatc", 4.56f);
+                put("doublec", 2.3445);
+                put("stringc", "test");
+                put("varcharc", "test2");
+                put("charc", 'c');
+                put("binaryc", new byte[]{ (byte)1 });
+                put("datec", new java.sql.Date(currentTimeMillis));
+                put("timestampc", new java.sql.Timestamp(currentTimeMillis));
+                put("decimalc", 0.45);
+            }
+        })));
+        assert(deserialized instanceof  List);
+        Date date = new Date();
+        date.setTimeInMillis(currentTimeMillis);
+        Timestamp ts = new Timestamp();
+        ts.setTimeInMillis(currentTimeMillis);
+
+        List<Object> fields = (List<Object>)deserialized;
+        assertEquals(Byte.valueOf("2"), fields.get(0));
+        assertEquals(Short.valueOf("45"), fields.get(1));
+        assertEquals(Integer.valueOf(95), fields.get(2));
+        assertEquals(Long.valueOf(876L), fields.get(3));
+        assertEquals(Boolean.TRUE, fields.get(4));
+        assertEquals(4.56f, fields.get(5));
+        assertEquals(2.3445, fields.get(6));
+        assertEquals("test", fields.get(7));
+        assertEquals("test2", fields.get(8));
+        assertEquals("c", fields.get(9));
+        //assertEquals(AvroTypeUtil.convertByteArray(new Object[]{ (byte)1 }).array(), (byte[])fields.get(10));
```

Comment:
Should this be included or removed?

Reply:
I'm thinking it probably should, would have caught the bug you saw